include/zephyr/net/ocpp.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file ocpp.h
+ *
+ * @defgroup ocpp_api OCPP library
+ * @ingroup networking
+ * @{
+ * @brief OCPP Charge Point Implementation
+ *
+ * @note The implementation assumes Websocket module is enabled.
+ *
+ * @note By default the implementation uses OCPP version 1.6.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_OCPP_H
+#define ZEPHYR_INCLUDE_NET_OCPP_H
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_ip.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CISTR50	50
+
+/**
+ * @brief OCPP IdTage authorization status in result to ocpp request
+ *        authorization
+ */
+typedef enum {
+	AUTH_ACCEPTED,
+	AUTH_BLOCKED,
+	AUTH_EXPIRED,
+	AUTH_INVALID,
+	AUTH_CONCURRENT_TX
+}auth_status_t;
+
+typedef enum {
+	/** user must fill the current reading */
+	OCPP_USR_GET_METER_VALUE,
+
+	/** process start charging request as like idtag received from local
+	  * e.g authorize etc
+	  */
+	OCPP_USR_START_CHARGING,
+
+	/** process stop charging sequence */
+	OCPP_USR_STOP_CHARGING,
+
+	/** unlock mechanical connector */
+	OCPP_USR_UNLOCK_CONNECTOR,
+}ocpp_notify_reason_t;
+
+/** @brief OCPP meter readings to be filled on user callback request from
+ *         library
+ */
+typedef enum {
+	OMM_CURRENT_FROM_EV,		// in A
+	OMM_CURRENT_TO_EV, 		// in A
+	OMM_CURRENT_MAX_OFFERED_TO_EV,	// in A
+	OMM_ACTIVE_ENERGY_FROM_EV,	// in Wh
+	OMM_ACTIVE_ENERGY_TO_EV,	// in Wh
+	OMM_REACTIVE_ENERGY_FROM_EV,	// in varh
+	OMM_REACTIVE_ENERGY_TO_EV,	// in varh
+	OMM_ACTIVE_POWER_FROM_EV,	// in W
+	OMM_ACTIVE_POWER_TO_EV,		// in W
+	OMM_REACTIVE_POWER_FROM_EV,	// in var
+	OMM_REACTIVE_POWER_TO_EV,	// in var
+	OMM_POWERLINE_FREQ,		//Hz
+	OMM_POWER_FACTOR,
+	OMM_POWER_MAX_OFFERED_TO_EV,	// in W
+	OMM_FAN_SPEED,			// in rpm
+	OMM_CHARGING_PERCENT,
+	OMM_TEMPERATURE,		// in Celsius
+	OMM_VOLTAGE_AC_RMS,		// in V
+
+	OMM_END
+}ocpp_meter_measurand_t;
+
+/** @brief OCPP user callback notification/request of input/output values
+ *         union member should be accessed with enum value ocpp_notify_reason_t
+ *         correspondingly.
+ *         e.g. callback reason is OCPP_USR_GET_METER_VALUE, struct meter_val
+ *              is valid
+ */
+typedef union {
+
+	struct {
+		/** input to user, requested connector_id or 0 - main meter */
+		int id_con;
+
+		ocpp_meter_measurand_t mes;
```

Comment:
All the variables should be documented by default. If some variables are kind of "private", you can add documentation guards `/** @cond INTERNAL_HIDDEN */` to prevent them from showing in the documentation, see examples in other header files.